native/effektio/api.rsh

```diff
@@ -94,6 +94,45 @@ object RoomMessage {
 
     /// the server receiving timestamp
     fn origin_server_ts() -> u64;
+
+    /// m.audio, m.emote, m.file, m.image, m.location, m.service_notice, m.text, m.video or m.key.verification.request
+    fn msgtype() -> string;
+
+    /// contains source data, name, mimetype, size, width and height
+    fn image_description() -> Result<ImageDescription>;
+
+    /// contains source data, name, mimetype and size
+    fn file_description() -> Result<FileDescription>;
```

Comment:
What if there is no image/file? I assume this is then `None`? Shouldn't this be `Option` rather than `Result` then?

Reply:
In fact, I was planning to user these functions for only file/image.
Will fix.

Reply:
I wanted to abstract image description and file description into a single struct.
But they have different fields, so I couldn't abstract them.